### ISSUE TITLE
Missing MSVC Redistributables

### ISSUE THREAD

**Issue by Zombeaver** [dosbox-staging/dosbox-staging#672]
**Describe the bug**
There are a couple redistributables that staging requires that aren't included in the download and aren't present by default in Windows, even in 10 (which is where I first encountered the issue). They're included in Visual Studio.

The specific dependencies are:

**x64:**
msvcp140.dll
vcruntime140.dll
vcruntime140_1.dll

**x86:**
msvcp140.dll
vcruntime140.dll

These can be deployed locally alongside staging and it works fine.

**To Reproduce**
Start DOSBox-staging on a computer that doesn't have the above dependencies.

**Expected behaviour**
For DOSBox-staging not to crash on launch.

**Environment:**
Windows 7/10

**Version of dosbox-staging:**
Any (though I was using v0.76.0-alpha-704-gce53 at the time)

**Comment by MasterO2** [comment]
Microsoft provides them here:

https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads

@kcgen previously gave me that link and it worked for me when I ran dosbox-staging 64-bit in Windows 10.

**Comment by Zombeaver** [comment]
Yes, I'm aware. As I said, I already have it working. The issue is that they're not included with staging itself. The dependencies should be included along with staging and aren't currently. I talked to @dreamer about it on Discord and he wanted me to submit an issue for it.

**Comment by MasterO2** [comment]
@kcgen @dreamer 

If MSVC redistributables are included in dosbox-staging 0.76 and beyond, will that cause issues for people who have already installed the MSVC redistributables?

**Comment by dreamer** [comment]
@MasterO2 no, it shouldn't cause any problems.

BTW, when we'll fix it, then fix is very likely to be backported to 0.75.x branch and trigger 0.75.3 release.

**Comment by dreamer** [comment]
@Zombeaver review for fix just started. Test builds inluding these files  will be available in a moment here:
https://github.com/dosbox-staging/dosbox-staging/actions/runs/315466660

Can you check if these work correctly for you?

Some technical/legal notes:

- Microsoft provides regular updates to those files, ATM we're bundling version 14.27.29016 (the newest available, as of today); this is unfortunate, as it adds new packaging step to our releases :(
- I tracked the exact license basis allowing us to bundle those files, made a summary in file [docs/vc_redist.txt](https://github.com/dosbox-staging/dosbox-staging/compare/po/vs-redist-2#diff-a3e174914c7753fc86cf4b432fe0333d40ef0bdc66859133667070ed6c14e49c) (bundled in the zip file).
- That being said, as of 2020 this license still permits us to distribute them (as it's "VS C++ 2015-2019 runtime package"); we are ok to distribute these under "MICROSOFT VISUAL STUDIO 2015 EXTENSIONS, VISUAL STUDIO SHELLS and C++ REDISTRIBUTABLE" licenses from 2015 and 2017... **BUT Microsoft REMOVED this provision from 2019 license**. We can expect, that once Windows 8.2 will reach EOL, distribution of new versions of these files will likely no longer be allowed. If/when that happens, we will rather stop distributing these instead of providing outdated versions (but as Windows 8.x will no longer be supported, all Windows 10 users should be covered by the built-in versions of these files by that time). Or we will provide alternative mechanism for installing them (I think there's a possibility to deploy them automatically when software is packaged in MSI file).

Until then, we will bump the version of these files once every few months (I think they are being updated in sync with Visual Studio updates, or perhaps with half-year schedule of Windows 10 versions - I'm not sure; ATM Windows images on GitHub CI provide 3 versions of those files: `14.16.27012`, `14.25.28508`, `14.27.29016`.

**Comment by Zombeaver** [comment]
Okay, sounds good. I'll test tonight and report back.

**Comment by Zombeaver** [comment]
Did some testing, with an old and current version for verification:

x86 alpha-704: crash
x86 alpha-851: no crash
x64 alpha-704: crash
x64 alpha-851: no crash

:smile:

**Comment by kcgen** [comment]
:thinking: I'm seeing a pattern here .... :-)
